frontend/middleware_logging.go

```diff
@@ -76,52 +76,67 @@ func MiddlewareLogging(w http.ResponseWriter, r *http.Request, next http.Handler
 }
 
 func MiddlewareLoggingPostMux(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
-	var pathValue string
-
 	ctx := r.Context()
 
 	correlationData := arm.NewCorrelationData(r)
 	ctx = ContextWithCorrelationData(ctx, correlationData)
 
-	w.Header().Set(arm.HeaderNameRequestID, correlationData.RequestID.String())
+	setHeaders(w, r, correlationData)
+
+	attrs := getLogAttrs(correlationData, r)
+
+	logger, err := LoggerFromContext(ctx)
+	if err != nil {
+		DefaultLogger().Error(err.Error())
+		arm.WriteInternalServerError(w)
+		return
+	}
+
+	handler := logger.Handler()
+	loggerWithAttrs := slog.New(handler.WithAttrs(attrs))
+	ctx = ContextWithLogger(ctx, loggerWithAttrs)
+
+	reqWithContext := r.WithContext(ctx)
+
+	next(w, reqWithContext)
+}
+
+// setHeaders writes the appropriate headers in the response writer
+// based on the request and the correlation data.
+func setHeaders(w http.ResponseWriter, r *http.Request, correlationData *arm.CorrelationData) {
+	w.Header().Set(arm.HeaderNameRequestID, correlationData.OperationID.String())
 
 	if strings.EqualFold(r.Header.Get(arm.HeaderNameReturnClientRequestID), "true") {
 		w.Header().Set(arm.HeaderNameClientRequestID, correlationData.ClientRequestID)
 	}
+}
 
+// getLogAttrs returns the appropiate Logging Attributes based on correlationData and a request.
+func getLogAttrs(correlationData *arm.CorrelationData, r *http.Request) []slog.Attr {
 	attrs := []slog.Attr{
-		slog.String("request_id", correlationData.RequestID.String()),
+		slog.String("request_id", correlationData.OperationID.String()),
 		slog.String("client_request_id", correlationData.ClientRequestID),
 		slog.String("correlation_request_id", correlationData.CorrelationRequestID),
 	}
 
+	var pathValue string
 	if pathValue = r.PathValue(PathSegmentSubscriptionID); pathValue != "" {
 		attrs = append(attrs, slog.String("subscription_id", pathValue))
 	}
-
 	if pathValue = r.PathValue(PathSegmentResourceGroupName); pathValue != "" {
 		attrs = append(attrs, slog.String("resource_group", pathValue))
 	}
 
 	if pathValue = r.PathValue(PathSegmentResourceName); pathValue != "" {
 		attrs = append(attrs, slog.String("resource_name", pathValue))
-		resource_id := fmt.Sprintf("/subscriptions/%s/resourcegroups/%s/providers/%s/%s",
+
+		resource_id := fmt.Sprintf(
+			"/subscriptions/%s/resourcegroups/%s/providers/%s/%s",
 			r.PathValue(PathSegmentSubscriptionID),
 			r.PathValue(PathSegmentResourceGroupName),
 			api.ResourceType,
 			pathValue)
```

Comment:
Question for the ARO HCP team: is it possible that the last "if" condition is wrong? Because in that "if", we just check if _PathSegmentResourceName_ is present, but later we construct the resourceID using _PathSegmentSubscriptionID_ and _PathSegmentResourceGroupName_ which we don't know if they are present in the URL as we have not checked them previously in the "if" (and hence they will return empty string, which will produce a resource ID with the value "/subscriptions//resourcegroups//providers/Microsoft.RedHatOpenShift/hcpOpenShiftClusters/the_resource_name" as shown in frontend/middleware_logging_test.go.

Reply:
+1 to refactoring this. We should just have a separate condition to check the whole path, then if the whole path exists append the `resource_id` field.

Reply:
Worth noting that we have full control over which endpoints that `MiddlewareLoggingPostMux` runs on (see `NewFrontend` in `frontend.go`).  Before we add too much more complexity here it might be worth re-evaluating if this middleware makes sense for endpoints that don't produce a full resource ID, like "list-by" endpoints.

Reply:
I agree with both of you. I have added the logic in the function to handle that as I don't like functions to make assumptions about external factors, and it is just a small check, we can always later revisit the usage of the middleware. Thanks!